apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/servlet/ServletApiAdvice.java

```diff
@@ -148,7 +148,13 @@ public static void onExitServletService(@Advice.Argument(0) ServletRequest servl
                     resp.addHeader(headerName, response.getHeader(headerName));
                 }
 
-                servletTransactionHelper.onAfter(transaction, t, response.isCommitted(), response.getStatus(), request.getMethod(),
+                int statusCode = response.getStatus();
```

Comment:
As this code is not Servlet API specific, I'd move this to ServletTransactionHelper